### PR TITLE
Use bidirectional A* for bicycle over 10km.

### PR DESCRIPTION
**Pull request by dnesbitt61**
Fix destination in bidirectional A* (needed to add opposing edge to the adjacency list). Update to TransitionCostReverse call. 

Depends on sif changes

### CODE REVIEW

**Comment by dgearhart** [comment]
:shipit:

**Comment by gknisely** [comment]
:bike:

File: src/thor/service.cc [hunk 1]
```diff
@@ -82,8 +82,8 @@ namespace {
           thor::PathAlgorithm* path_algorithm;
           if (costing == "multimodal") {
             path_algorithm = &multi_modal_astar;
-          } else if (costing == "pedestrian") {
-            // Use bidirectional A* if over 10km
+          } else if (costing == "pedestrian" || costing == "bicycle") {
+            // Use bidirectional A* for pedestrian and bicycle if over 10km
             float dist = origin.latlng_.Distance(destination.latlng_);
             path_algorithm = (dist > 10000.0f) ? &bidir_astar : &astar;
```
Comment (kevinkreiser) [inline src/thor/service.cc]: is 10k the point at which performance is normally better with bidirectional? less than that unidirectional has less setup and bookkeeping and so is more efficient? just wondering if its empirically derived

**Comment by kevinkreiser** [comment]
:see_no_evil:

File: src/thor/service.cc (hunk 1 above)
Comment (dnesbitt61) [inline src/thor/service.cc]: It has more to do with making sure the special cases that the single tree A* handles (loop edge) are taken care of. We could test bidirectional for all these edge cases that can occur on short routes and make sure it works - then we could remove the distance check.

Comment (kevinkreiser) [inline src/thor/service.cc]: ah ok cool. this seems quite reasonable in the interim
